File: typescript/rna-transcription/rna-transcription.ts
const dna2rna: { [key: string]: string } = {
  G: 'C',
  C: 'G',
  T: 'A',
  A: 'U'
}

export function toRna( dna: string ): string {
  return [...dna].reduce((rna, nucleotide) => {
    if (nucleotide in dna2rna)
      return rna + dna2rna[nucleotide]
    else
      throw new Error('Invalid input DNA.')
  }, '')
}

File: lua/rna-transcription/rna-transcription.lua
local dna2rna = { G='C', C='G', T='A', A='U' }

local to_rna = function (dna)
    local rna = ""
    for char in dna:gmatch(".") do
        rna = rna .. dna2rna[char]
    end
    return rna
end

return to_rna
